### PR TITLE
Feat/eo consent between users

### PR DESCRIPTION
**Pull request by frontend-specialisten**
## Description

This pull request includes several changes to the `eo` library, focusing on improving the authentication and consent management features. The most important changes include the addition of a new guard, modifications to the actor service, updates to the consent service, and enhancements to the user interface components.

### Authentication Enhancements:
* Added a new guard `eoActorSelfGuard` to ensure users are acting as themselves before accessing certain routes (`libs/eo/auth/data-access/actor-self.guard.ts`).
* Updated `EoActorService` to simplify the `self` property initialization and fixed a typo in the method comment (`libs/eo/auth/data-access/actor.service.ts`). [[1]](diffhunk://#diff-ee28c3b3fa7918ea1713b353ad7af109763f5eaeb1c2e760bc4232aea84f94d0L38-R38) [[2]](diffhunk://#diff-ee28c3b3fa7918ea1713b353ad7af109763f5eaeb1c2e760bc4232aea84f94d0L57-R55)

### Consent Management Enhancements:
* Expanded `EoConsent` and `EoReceivedConsent` interfaces to include additional organization-related fields (`libs/eo/consent/data-access-api/consent.service.ts`).
* Updated `EoConsentService` with new methods for handling organization consents and modified existing methods to accommodate the new consent structure (`libs/eo/consent/data-access-api/consent.service.ts`). [[1]](diffhunk://#diff-626c4ec91954769fc1f826c1018b03915bc830cf39382a2841bcc1aaf40bc04aR61-R66) [[2]](diffhunk://#diff-626c4ec91954769fc1f826c1018b03915bc830cf39382a2841bcc1aaf40bc04aL62-R92)

### User Interface Improvements:
* Enhanced `EoActorMenuComponent` to display a checkmark next to the currently selected actor and adjusted the layout for better user experience (`libs/eo/auth/ui-actor-menu/actor-menu.component.ts`). [[1]](diffhunk://#diff-d9a31f666ad985d75ce234c4b5b7270514f05f5c68c3cb5ab7634e12c2f0666cR99-R100) [[2]](diffhunk://#diff-d9a31f666ad985d75ce234c4b5b7270514f05f5c68c3cb5ab7634e12c2f0666cL138-R140) [[3]](diffhunk://#diff-d9a31f666ad985d75ce234c4b5b7270514f05f5c68c3cb5ab7634e12c2f0666cL153) [[4]](diffhunk://#diff-d9a31f666ad985d75ce234c4b5b7270514f05f5c68c3cb5ab7634e12c2f0666cL162-R189)
* Updated `EoConsentDetailsDrawerComponent` and `EoEditConsentModalComponent` to use the new organization fields in the consent details (`libs/eo/consent/feature-details/consent-details.component.ts`, `libs/eo/consent/feature-edit-consent/edit-consent.component.ts`). [[1]](diffhunk://#diff-2fd5902ab91c070417534ce88a602163c9e8f8b1596035f22888aa25e63e9d06L108-R108) [[2]](diffhunk://#diff-2fd5902ab91c070417534ce88a602163c9e8f8b1596035f22888aa25e63e9d06L124-R124) [[3]](diffhunk://#diff-e0e4faf4932a2af22a9ecf7f119fc78e1b3285e9f968516dd3ff76a67bdeb305L107-R123) [[4]](diffhunk://#diff-e0e4faf4932a2af22a9ecf7f119fc78e1b3285e9f968516dd3ff76a67bdeb305L186-R186)
* Improved `EoGrantConsentModalComponent` to handle both client and organization consents (`libs/eo/consent/feature-grant-consent/grant-consent.component.ts`). [[1]](diffhunk://#diff-ab51dff7f69edd9753bfa8b1bc2ab5c0c1a2cd2cd63fb603f3a5f65b85721f69L164-R166) [[2]](diffhunk://#diff-ab51dff7f69edd9753bfa8b1bc2ab5c0c1a2cd2cd63fb603f3a5f65b85721f69L177-R215)

### Miscellaneous:
* Added `EoDataIntlService` to customize internationalization for data components and integrated it into `EoConsentOverviewComponent` (`libs/eo/consent/feature-overview/consent-overview.component.ts`). [[1]](diffhunk://#diff-3e3bef5762e27a5f84ee6ec47c1e114fa6d08b5380ec73067a16f0695fdd0c3dL17-R22) [[2]](diffhunk://#diff-3e3bef5762e27a5f84ee6ec47c1e114fa6d08b5380ec73067a16f0695fdd0c3dL35-R66) [[3]](diffhunk://#diff-3e3bef5762e27a5f84ee6ec47c1e114fa6d08b5380ec73067a16f0695fdd0c3dL58-R82) [[4]](diffhunk://#diff-3e3bef5762e27a5f84ee6ec47c1e114fa6d08b5380ec73067a16f0695fdd0c3dL75-R95) [[5]](diffhunk://#diff-3e3bef5762e27a5f84ee6ec47c1e114fa6d08b5380ec73067a16f0695fdd0c3dR111-R116) [[6]](diffhunk://#diff-3e3bef5762e27a5f84ee6ec47c1e114fa6d08b5380ec73067a16f0695fdd0c3dR129-L110) [[7]](diffhunk://#diff-3e3bef5762e27a5f84ee6ec47c1e114fa6d08b5380ec73067a16f0695fdd0c3dR144-R146) [[8]](diffhunk://#diff-3e3bef5762e27a5f84ee6ec47c1e114fa6d08b5380ec73067a16f0695fdd0c3dR160-R163)
* Exported the new guard from the data-access index (`libs/eo/auth/data-access/index.ts`).

## References

- [#3280](https://github.com/Energinet-DataHub/energy-origin-issues/issues/3280)
- [#3568](https://github.com/Energinet-DataHub/energy-origin-issues/issues/3280)

### CODE REVIEW

File: libs/eo/consent/feature-grant-consent/grant-consent.component.ts [hunk 1]
```diff
@@ -174,24 +175,44 @@ export class EoGrantConsentModalComponent {
   public opened = false;
 
   open() {
-    // If the third party client id is not set, theres no reason to open the modal
-    if (!this.thirdPartyClientId) return;
+    // If the third party client id or organization id is not set, theres no reason to open the modal
+    if (!this.thirdPartyClientId && !this.organizationId) return;
```
Comment (sahma19) [inline libs/eo/consent/feature-grant-consent/grant-consent.component.ts:180]: I hvilken situation kan dette ske?

File: libs/eo/consent/feature-grant-consent/grant-consent.component.ts [hunk 2]
```diff
@@ -174,24 +175,44 @@ export class EoGrantConsentModalComponent {
...
+
+    if(this.thirdPartyClientId) {
+      grantConsentRequest = this.consentService.grantClient(this.thirdPartyClientId);
+    }
+
+    if(this.organizationId) {
+      grantConsentRequest = this.consentService.grantOrganization(this.organizationId);
+    }
+
+    grantConsentRequest?.subscribe({
```
Comment (sahma19) [inline libs/eo/consent/feature-grant-consent/grant-consent.component.ts:215]: Kan begge if statements være true, eller kun 1 af dem?

File: libs/eo/core/globalization/assets-localization/i18n/da.ts [hunk 3]
```diff
@@ -744,18 +744,18 @@ export const DA_TRANSLATIONS: TranslationKeys = {
...
+      <h4>Tilbagekaldelse af fuldmagt</h4>
+      <p>Fuldmagten kan til enhver tid tilbagekaldes, hvis deling af data med {{organizationName}} ikke længere ønskes. Dette kan gøres ved at klikke på 'Tilbagekald fuldmagt'.</p>
+      <p>Ved tilbagekaldelse vil {{organizationName}} ikke længere kunne hente data og administrere certifikater.</p>
     `,
-    cancel: 'Cancel',
-    saveChanges: 'Save changes',
-    revoke: 'Revoke power of attorney',
-    revokeSuccess: 'The power of attorney has been revoked',
-    revokeError: 'An error occurred while revoking the power of attorney. Please try again.',
+    cancel: 'Annuller',
```
Comment (sahma19) [inline libs/eo/core/globalization/assets-localization/i18n/da.ts:754]:
```suggestion
    cancel: 'Annullér',
```

File: libs/eo/core/globalization/assets-localization/i18n/da.ts [hunk 4]
```diff
@@ -744,18 +744,18 @@ export const DA_TRANSLATIONS: TranslationKeys = {
...
+    saveChanges: 'Gem ændringer',
+    revoke: 'Tilbagekald fuldmagt',
+    revokeSuccess: 'Fuldmagten er blevet tilbagekaldt',
+    revokeError: 'Der opstod en fejl ved tilbagekaldelse af fuldmagten. Prøv igen.',
```
Comment (sahma19) [inline libs/eo/core/globalization/assets-localization/i18n/da.ts:758]:
```suggestion
    revokeError: 'Der opstod en fejl ved tilbagekaldelse af fuldmagten. Prøv venligst igen.',
```

File: libs/eo/core/globalization/assets-localization/i18n/da.ts [hunk 5]
```diff
@@ -807,6 +807,18 @@ export const DA_TRANSLATIONS: TranslationKeys = {
       message: 'Der opstod en uventet fejl. Prøv igen.',
     },
   },
+  requestConsent: {
+    title: 'Anmod om fuldmagt',
+    description: `
+        <h4>Sådan anmoder du om fuldmagt</h4>
+        <ol>
+          <li>Kopier linket og send det til den, du ønsker fuldmagt af (send kun linket til en modtager)</li>
```
Comment (sahma19) [inline libs/eo/core/globalization/assets-localization/i18n/da.ts:815]:
```suggestion
          <li>Kopier linket og send det til den du ønsker fuldmagt fra (send kun linket til én modtager)</li>
```

File: libs/eo/core/globalization/assets-localization/i18n/da.ts [hunk 6]
```diff
@@ -807,6 +807,18 @@ export const DA_TRANSLATIONS: TranslationKeys = {
...
+          <li>Modtager klikker på linket, logger på energioprindelse og afgiver fuldmagt</li>
```
Comment (sahma19) [inline libs/eo/core/globalization/assets-localization/i18n/da.ts:816]:
```suggestion
          <li>Modtager trykker på linket, logger på energytrackandtrace.dk, og afgiver fuldmagt</li>
```

File: libs/eo/core/globalization/assets-localization/i18n/en.ts [hunk 7]
```diff
@@ -820,6 +820,18 @@ export const EN_TRANSLATIONS: TranslationKeys = {
       message: 'An unexpected error occurred. Please try again.',
     },
   },
+  requestConsent: {
+    title: 'Request Power of Attorney',
+    description: `
+      <h4>How to request Power of Attorney</h4>
+      <ol>
+        <li>Copy the link and send it to the person you want Power of Attorney from (send the link to one recipient only)</li>
```
Comment (sahma19) [inline libs/eo/core/globalization/assets-localization/i18n/en.ts:828]:
```suggestion
        <li>Copy the link, and send it to the person you want Power of Attorney from. (Make sure to send the link to one recipient, only)</li>
```

File: libs/eo/core/globalization/assets-localization/i18n/en.ts [hunk 8]
```diff
@@ -820,6 +820,18 @@ export const EN_TRANSLATIONS: TranslationKeys = {
...
+        <li>The recipient clicks the link, logs into energyorigin and grants Power of Attorney</li>
```
Comment (sahma19) [inline libs/eo/core/globalization/assets-localization/i18n/en.ts:829]:
```suggestion
        <li>The recipient clicks the link, logs into energytrackandtrace.dk, and grants Power of Attorney</li>
```

File: libs/eo/shared/assets/src/assets/configuration/eo-api-environment.local.json [hunk 9]
```diff
@@ -1,5 +1,5 @@
 {
-  "apiBase": "https://demo.energytrackandtrace.dk/api",
+  "apiBase": "https://energytrackandtrace-dk-addorganizationconsent.v.acorn-dev.dk/api",
```
Comment (sahma19) [inline libs/eo/shared/assets/src/assets/configuration/eo-api-environment.local.json:2]: Skal den forblive sådan?

File: libs/eo/shared/assets/src/assets/configuration/eo-azure-b2c-settings.local.json [hunk 10]
```diff
@@ -1,6 +1,6 @@
 {
   "azure-b2c": {
-    "issuer": "https://datahubeouenerginet.b2clogin.com/datahubeouenerginet.onmicrosoft.com/B2C_1A_OIDCMOCK/v2.0/",
+    "issuer": "https://datahubeouenerginet.b2clogin.com/datahubeouenerginet.onmicrosoft.com/B2C_1A_OIDCMOCK_CONSENT_TEST/v2.0/",
```
Comment (sahma19) [inline libs/eo/shared/assets/src/assets/configuration/eo-azure-b2c-settings.local.json:3]: Skal den forblive sådan?

File: libs/eo/consent/feature-grant-consent/grant-consent.component.ts (hunk 2 above)
Comment (frontend-specialisten) [reply libs/eo/consent/feature-grant-consent/grant-consent.component.ts]: Kun den ene, det er nu rettet til en else

File: libs/eo/shared/assets/src/assets/configuration/eo-api-environment.local.json (hunk 9 above)
Comment (frontend-specialisten) [reply libs/eo/shared/assets/src/assets/configuration/eo-api-environment.local.json]: reverted

File: libs/eo/shared/assets/src/assets/configuration/eo-azure-b2c-settings.local.json (hunk 10 above)
Comment (frontend-specialisten) [reply libs/eo/shared/assets/src/assets/configuration/eo-azure-b2c-settings.local.json]: reverted

File: libs/eo/consent/feature-grant-consent/grant-consent.component.ts (hunk 1 above)
Comment (sahma19) [reply libs/eo/consent/feature-grant-consent/grant-consent.component.ts:180]: Men hvad ville der ske, hvis du ikke returnede her? ville du så crashe applikationen indtil den bliver restarted?

**Review by sahma19 — APPROVED** [review]

**Review by sahma19 — APPROVED** [review]

**Pull request merged**